front/components/vaults/VaultDataSourceViewContentList.tsx

```diff
@@ -55,39 +59,53 @@ type VaultDataSourceViewContentListProps = {
   connector: ConnectorType | null;
 };
 
-const getTableColumns = (): ColumnDef<RowData, string>[] => {
-  const columns: ColumnDef<RowData, string>[] = [
-    {
-      header: "Name",
-      accessorKey: "title",
-      id: "title",
-      sortingFn: "text", // built-in sorting function case-insensitive
-      cell: (info: CellContext<RowData, unknown>) => (
-        <DataTable.CellContent icon={info.row.original.icon}>
-          <span>{info.row.original.title}</span>
+const getTableColumns = (showVaultUsage: boolean): ColumnDef<RowData>[] => {
+  const columns: ColumnDef<RowData, any>[] = [];
+  columns.push({
+    header: "Name",
+    accessorKey: "title",
+    id: "title",
+    sortingFn: "text", // built-in sorting function case-insensitive
+    cell: (info: CellContext<RowData, string>) => (
+      <DataTable.CellContent icon={info.row.original.icon}>
+        <span>{info.getValue()}</span>
+      </DataTable.CellContent>
+    ),
+  });
+
+  if (showVaultUsage) {
+    columns.push({
+      header: "Available to",
+      accessorKey: "vaults",
+      cell: (info: CellContext<RowData, VaultType[]>) => (
+        <DataTable.CellContent>
+          {info
+            .getValue()
+            .map((v) => v.name)
+            .join(",")}
```

Comment:
IIRC we have a util function to that crops too long string, could be useful here

Reply:
Maybe let's add a "-" if the list is empty?